flink-table-store-connector/src/test/java/org/apache/flink/table/store/connector/FileStoreITCase.java

```diff
@@ -0,0 +1,210 @@
+
+import static org.apache.flink.table.store.file.FileStoreOptions.BUCKET;
+import static org.apache.flink.table.store.file.FileStoreOptions.FILE_FORMAT;
+import static org.apache.flink.table.store.file.FileStoreOptions.FILE_PATH;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/** ITCase for {@link FileStoreSource} and {@link StoreSink}. */
+@RunWith(Parameterized.class)
+public class FileStoreITCase extends AbstractTestBase {
+
+    private static final RowType PARTITION_TYPE =
+            new RowType(Collections.singletonList(new RowType.RowField("p", new VarCharType())));
+
+    private static final RowType KEY_TYPE =
+            new RowType(Collections.singletonList(new RowType.RowField("k", new IntType())));
+
+    private static final RowType VALUE_TYPE =
+            new RowType(
+                    Arrays.asList(
+                            new RowType.RowField("v", new IntType()),
+                            new RowType.RowField("p", new VarCharType()),
+                            // rename key
+                            new RowType.RowField("_k", new IntType())));
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    private static final DataStructureConverter<RowData, Row> CONVERTER =
+            (DataStructureConverter)
+                    DataStructureConverters.getConverter(
+                            TypeConversions.fromLogicalToDataType(VALUE_TYPE));
+
+    private static final int NUM_BUCKET = 3;
+
+    private static final List<RowData> SOURCE_DATA =
+            Arrays.asList(
+                    wrap(GenericRowData.of(0, StringData.fromString("p1"), 1)),
+                    wrap(GenericRowData.of(0, StringData.fromString("p1"), 2)),
+                    wrap(GenericRowData.of(5, StringData.fromString("p1"), 1)),
+                    wrap(GenericRowData.of(6, StringData.fromString("p2"), 1)),
+                    wrap(GenericRowData.of(3, StringData.fromString("p2"), 5)),
+                    wrap(GenericRowData.of(5, StringData.fromString("p2"), 1)));
+
+    private File tempPath;
+
+    private Configuration options;
+
+    private final boolean isBounded;
+
+    public FileStoreITCase(boolean isBounded) {
+        this.isBounded = isBounded;
+    }
+
+    @Parameterized.Parameters(name = "isBounded-{0}")
+    public static List<Boolean> getVarSeg() {
+        return Arrays.asList(true, false);
+    }
+
+    private static SerializableRowData wrap(RowData row) {
+        return new SerializableRowData(row, InternalSerializers.create(VALUE_TYPE));
+    }
+
+    @Before
+    public void before() throws IOException {
+        tempPath = TEMPORARY_FOLDER.newFolder();
+        options = new Configuration();
+        options.set(BUCKET, NUM_BUCKET);
+        options.set(FILE_PATH, tempPath.toURI().toString());
+        options.set(FILE_FORMAT, "avro");
+    }
+
+    @SuppressWarnings("ResultOfMethodCallIgnored")
+    @After
+    public void after() {
+        if (tempPath != null) {
```

Comment:
Nit: delete `tempPath` here is somewhat redundant because `TEMPORARY_FOLDER.newFolder()` will generate different subdirectories for each test, and class rule `TEMPORARY_FOLDER` will delete the parent directory and its content recursively at the end.